r-package/grf/R/input_utilities.R

```diff
@@ -77,6 +77,9 @@ validate_seed <- function(seed) {
 }
 
 validate_clusters <- function(clusters, X) {
+  # This will convert factor to integer
+  clusters <- unclass(clusters)
```

Comment:
I find this use of `unclass` tricky to follow, and I haven't seen it used before in this context. Here is a simple suggestion that would work and is fairly readable:
```
...
} else if (!is.vector(clusters) || !(is.numeric(clusters) || is.factor(clusters))) {
     stop("Clusters must be a vector of integers or factors.")
} else if (is.numeric(clusters) && !all(clusters == floor(clusters))) {
    stop("Clusters cannot contain floating point values.")
}
```

Reply:
The above (and my most recent commit) does not work as `is.vector(iris$Species) == FALSE`. `is.vector` will only return `TRUE` if there're no attributes (including `class`).

Can I ask what the motivation for using `is.vector` is here? There might be a better way to exclude what you're trying to exclude. From what I can tell, `mode(clusters) == "numeric"`, should be sufficient to include only inputs that can be coerced to `integer` or `double`. This will also extend allowable inputs to dates and other reasonable inputs for this variable. `mode(clusters) == "numeric"` should exclude only `lists` (including `S3`) and more complicated classes (like `S4`).

Reply:
Thanks, I had overlooked this part. I agree, I think we can just remove the `is.vector` check here.

Reply:
OK - I have updated the PR with the `mode(clusters)` test